File: web-sources/S8-pkgs-ui-paper/form/PaperForm.js
import { S8 } from "/S8-api/S8Context.js";
import { S8Object } from "/S8-api/S8Object.js";


import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";

import { PaperFormField } from "./PaperFormField.js";


S8.page.CSS_import("/S8-pkgs-ui-paper/form/PaperForm.css");


/**
 * 
 */
export class PaperForm extends S8Object {

    /**
     * @type{HTMLDivElement}
     */
    wrapperNode;

    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("paperform-wrapper");

        this.coreNode = document.createElement("div");
        this.coreNode.classList.add("paperform-core");
        this.wrapperNode.appendChild(this.coreNode);
    }


    getEnvelope(){ return this.wrapperNode; }

    /**
     * 
     * @param {PaperFormField[]} fields 
     */
    S8_set_fields(fields) {
        S8WebFront.setS8Elements(this.coreNode, fields);
    }



    S8_render(){ /* nothing to do */ }
    S8_dispose(){ /* nothing to do */ }
}

File: sources/com/s8/pkgs/ui/paper/form/PaperForm.java
package com.s8.pkgs.ui.paper.form;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.pkgs.ui.paper.PaperWebSources;


/**
 * 
 */
public class PaperForm extends S8WebObject {
	
	
	public PaperForm(S8WebFront front) {
		super(front, PaperWebSources.ROOT_WEBPATH + "/form/PaperForm");
	}
	

	
	/**
	 * 
	 * @param options
	 */
	public void setFields(PaperFormField[] fields) {
		vertex.outbound().setObjectListField("fields", fields);
	}
	
	

	
}
